### PR TITLE
Add accounts.authority_id to serve as a new unique account identifier.

### PR DESCRIPTION
**Pull request by shanemc-goog**
### Description
Adds a new authority_id field to Accounts to serve as a unique ID. Email is not unique in OIDC but that has been assumed and relied on.

Part of addressing #1793

### CODE REVIEW

**Comment by shanemc-goog** [comment]
This adds the DB table and accessors for the new account id.

**Review by bion — APPROVED** [review]

File: universal-application-tool-0.0.1/app/models/Account.java [hunk 1]
```diff
@@ -21,6 +21,10 @@
  * <p>When a user logs in for the first time either using SSO or as a guest, CiviForm creates an
  * {@code Account} record for them.
  *
+ * <p>emailAddress serves as the unchanging unique identifier for accounts though it is not
+ * gauranteed to not change in authentication protocols like OIDC. When #1793 is resolved though
```
Comment (zaralouis-sf) [inline universal-application-tool-0.0.1/app/models/Account.java:25]: Very minor nit: Guaranteed

File: universal-application-tool-0.0.1/conf/evolutions/default/36.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# --- Add a new unique identifier for accounts based on the authentication systems unique ID semantics.
+# --- Until we fully migrate from email_address to the new authority_id as the unique account ID both will identify
+# --- accounts, first with authority_id if present, then eamil_address..
```
Comment (zaralouis-sf) [inline universal-application-tool-0.0.1/conf/evolutions/default/36.sql:3]: Very minor nit: email

**Review by zaralouis-sf — APPROVED** [review]

File: universal-application-tool-0.0.1/conf/evolutions/default/36.sql (hunk 2 above)
Comment (shanemc-goog) [reply universal-application-tool-0.0.1/conf/evolutions/default/36.sql]: done

File: universal-application-tool-0.0.1/app/models/Account.java (hunk 1 above)
Comment (shanemc-goog) [reply universal-application-tool-0.0.1/app/models/Account.java]: done, thank you!

**Pull request merged**